examples/echo-server.rs

```diff
@@ -41,6 +42,15 @@ async fn accept_connection(stream: TcpStream) {
 
     info!("New WebSocket connection: {}", addr);
 
-    let (write, read) = ws_stream.split();
-    read.forward(write).await.expect("Failed to forward message")
+    let (mut write, mut read) = ws_stream.split();
+    while let Some(msg) = read.next().await {
+        let resp = match msg.expect("Failed to read messages") {
+            msg @ (Message::Text(_) | Message::Binary(_)) => msg,
+            // Match Message::Close here.
+            // We should not send any message after the connection is closed.
+            Message::Close(_) => break,
+            _ => continue,
+        };
+        write.send(resp).await.expect("Failed to send messages");
+    }
```

Comment:
Sorry for the late review.

I think there is a bit more elegant way to write it preserving the previous `read.forward(write)` notation with combinators, e.g. something like `read.filter(|m| m.is_binary() || m.is_text()).forward(write)`.

NB: Though you'll have to use `try_filter` I believe because of the type of the `Stream`, but I think you got the point :)

Reply:
Fixed with an elegant way! It looks nicer now.